include/tvm/runtime/crt/rpc_common/buffer.h

```diff
@@ -0,0 +1,70 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+/*!
+ * \file tvm/runtime/crt/rpc_common/buffer.h
+ * \brief Defines a buffer implementation for the RPC layers.
+ */
+
+#ifndef TVM_RUNTIME_CRT_RPC_COMMON_BUFFER_H_
+#define TVM_RUNTIME_CRT_RPC_COMMON_BUFFER_H_
+
+#include <inttypes.h>
+#include <stdlib.h>
+
+namespace tvm {
+namespace runtime {
+
+class Buffer {
+ public:
+  Buffer(uint8_t* data, size_t data_size_bytes)
+      : data_{data}, capacity_{data_size_bytes}, num_valid_bytes_{0}, read_cursor_{0} {}
+
+  size_t Write(const uint8_t* data, size_t data_size_bytes);
+
+  size_t Read(uint8_t* data, size_t data_size_bytes);
+
+  size_t Peek(uint8_t* data, size_t data_size_bytes);
+
+  void Clear();
+
+  inline size_t ReadAvailable() const { return num_valid_bytes_ - read_cursor_; }
```

Comment:
C++ convention, if the function is defined inplace, then inline is not necessary